src/documenting/code-groups.md

```diff
@@ -0,0 +1,165 @@
+# Code Groups
+
+Iroha 2 Documentation includes a custom Markdown Extension in order to
+support code groups. It is achieved with `:::code-group` container.
+
+## Markdown Syntax
+
+**Input**
+
+````md
+:::code-group
+
+```ts
+type A = [string, string]
+```
+
+### I have a title
+
+```rust
+assert_eq!(vec![], vec![]);
+```
+
+```
+Fence without lang or title
+```
+
+### Snippet syntax is supported as well
+
+<<<@/snippets/debug_java_Lorem.java
+
+:::
+````
+
+**Output**
+
+:::code-group
+
+```ts
+type A = [string, string]
+```
+
+### I have a title
+
+```rust
+assert_eq!(vec![], vec![]);
+```
+
+```
+Fence without lang or title
+```
+
+### Snippet syntax is supported as well
+
+<<<@/snippets/debug_java_Lorem.java
+
+:::
+
+**Rules:**
```

Comment:
I'd put this before input-output example

Reply:
Done